src/astarte-client/client.ts

```diff
@@ -320,17 +329,49 @@ class AstarteClient {
     deviceId: AstarteDevice['id'];
     interfaceName: AstarteInterface['name'];
   }): Promise<AstarteInterfaceValues> {
-    const { deviceId, interfaceName } = params;
     const response = await this.$get(
       this.apiConfig.deviceData({
-        deviceId,
-        interfaceName,
+        deviceId: params.deviceId,
+        interfaceName: params.interfaceName,
         ...this.config,
       }),
     );
     return response.data;
   }
 
+  async getDeviceDataTree(
+    params: DeviceOrDeviceIdParams & InterfaceOrInterfaceNameParams,
+  ): Promise<AstarteDataTreeNode> {
+    let device: AstarteDevice;
+    let iface: AstarteInterface;
+    if ('device' in params) {
+      device = params.device;
+    } else {
+      device = await this.getDeviceInfo(params.deviceId).catch(() => {
+        throw new Error(`Could not find device with ID ${params.deviceId}`);
+      });
+    }
+    if ('interface' in params) {
+      iface = params.interface;
+    } else {
+      const interfaceIntrospection = device.introspection.get(params.interfaceName);
+      if (!interfaceIntrospection) {
+        throw new Error(`Could not find interface ${params.interfaceName} in device introspection`);
```

Comment:
this is connected to the previous message.
If you only have the interface name you must check the device introspection to retrieve the major used by the device, BUT if you require an interface definition in order to properly format the data, and indeed an interface is provided, you don't need to check the introspection.
if a malicious developer tries to retrieve data from an interface not supported using this method, the Astarte REST API will not return any data